File: SofiForce.Models/Models/EntityModels/SupervisorModel.cs
using System;

namespace Models
{
    public class SupervisorModel
    {
        public int SupervisorId { get; set; }
        public int? SupervisorTypeId { get; set; }
        public int? BusinessUnitId { get; set; }
        public string CompanyCode { get; set; }
        public string SupervisorCode { get; set; }
        public string SupervisorNameEn { get; set; }
        public string SupervisorNameAr { get; set; }
        public string Phone { get; set; }
        public string PhoneAlternative { get; set; }
        public DateTime? JoinDate { get; set; }
        public int? UserId { get; set; }
        public int? BranchId { get; set; }
        public bool? IsActive { get; set; }
        public string Color { get; set; }
        public int? DisplayOrder { get; set; }
        public string Icon { get; set; }
        public bool? CanEdit { get; set; }
        public bool? CanDelete { get; set; }
        public string Notes { get; set; }
        public bool? IsTerminated { get; set; }
        public DateTime? TerminationDate { get; set; }
        public int? TerminationReasonId { get; set; }
        public int? CBy { get; set; }
        public DateTime? CDate { get; set; }
        public int? EBy { get; set; }
        public DateTime? EDate { get; set; }
    }
}

File: SofiForce.Web/ClientApp/src/app/core/Models/EntityModels/SupervisorModel.ts
// import { SupervisorTypeModel } from "./supervisorTypeModel";
import { BranchModel } from "./branchModel";
import { RepresentativeModel } from "./representativeModel";

export class SupervisorModel {
    supervisorId: number;
    supervisorTypeId: number | null;
    supervisorCode: string;
    companyCode: string;
    supervisorNameEn: string;
    supervisorNameAr: string;
    joinDate: Date | null;
    userId: number | null;
    branchId: number | null;
    branchCode: string;
    isActive: boolean | null;
    color: string;
    displayOrder: number | null;
    icon: string;
    canEdit: boolean | null;
    canDelete: boolean | null;
    notes: string;
    cBy: number | null;
    cDate: Date | null;
    eBy: number | null;
    eDate: Date | null;

    isTerminated: boolean | null;
    phone : string | null;
    phoneAlternative : string | null;
    terminationDate : Date | null;
    terminationReasonId : number | null;
    businessUnitId: number | null;
}